MaterialDesignThemes.Wpf/PopupBox.cs

```diff
@@ -439,21 +453,67 @@ public override void OnApplyTemplate()
             VisualStateManager.GoToState(this, IsPopupOpen ? PopupIsOpenStateName : PopupIsClosedStateName, false);
         }
 
-        protected override void OnIsKeyboardFocusWithinChanged(DependencyPropertyChangedEventArgs e)
+        bool supressfirstElementGotFocus = true;
+        IInputElement firstchildelement = null;
+
+        protected override void OnGotKeyboardFocus(KeyboardFocusChangedEventArgs e)
         {
-            base.OnIsKeyboardFocusWithinChanged(e);
+            base.OnGotKeyboardFocus(e);
+            if (supressfirstElementGotFocus == false) { return; }
 
-            if (IsPopupOpen && !IsKeyboardFocusWithin && !StaysOpen)
+            if (!IsPopupOpen && (PopupMode == PopupBoxPopupMode.ClickOrFocusWithin | 
+                                 PopupMode == PopupBoxPopupMode.MouseOverEagerOrFocusWithin | PopupMode == PopupBoxPopupMode.MouseOverOrFocusWithin)
+                             && IsKeyboardFocusWithin)
```

Comment:
Why use the unary `|` here? I think `||` is the better fit.

Reply:
I've made a mistake here. I got confused which of the two were the short-circuit OR here. I won't forget anymore.

Fixed.